doc/document.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package doc
+
+import (
+	"bytes"
+	"sort"
+)
+
+// IDFieldName is the field name reserved for IDs.
+var IDFieldName = []byte("_id")
```

Comment:
nit: rename to indicate it's a reserved word. `ReservedFieldNameID` or something along the lines.

Assuming it's a var to allow library users to change, yea? if so, could you put a little note indicating that.

Reply:
sure, sounds good

and nah, it has to be a `var` if i want it to be a byte slice. I could make it a string and make it const but then I have to convert it to a byte slice whenever I want to use it.

Reply:
talked offline, changing value to `var ... = "_m3ninx_id" `